src/sage/modules/free_module_element.pyx

```diff
@@ -4179,14 +4181,15 @@ cdef class FreeModuleElement(Vector):   # abstract base class
         # return self.apply_map(lambda x: x.nintegral(*args, **kwds) for x in self)
 
         if self.is_sparse():
-            v = [(i,z.nintegral(*args, **kwds)) for i,z in self.dict(copy=False).items()]
-            answers = dict([(i,a[0]) for i,a in v])
-            v=dict(v)
+            v = [(i, z.nintegral(*args, **kwds))
+                 for i, z in self.dict(copy=False).items()]
+            answers = {i: a[0] for i, a in v}
```

Comment:
It seems like we could go straight to a dict for `v` here. Then `answers` would be `{i: v[i][0] for i in v}`.